pkg/retriever/bitswapretriever.go

```diff
@@ -126,15 +139,15 @@ func (br *bitswapRetrieval) RetrieveFromAsyncCandidates(ayncCandidates types.Inb
 		lastBytesReceivedTimer = br.clock.AfterFunc(br.cfg.BlockTimeout, cancel)
 	}
 
-	totalWritten := uint64(0)
-	blockCount := uint64(0)
+	totalWritten := atomic.Uint64{}
+	blockCount := atomic.Uint64{}
 	cb := func(bytesWritten uint64) {
 		// record first byte received
-		if totalWritten == 0 {
+		if totalWritten.Load() == 0 {
 			br.events(events.FirstByte(br.request.RetrievalID, phaseStartTime, bitswapCandidate))
 		}
-		atomic.AddUint64(&totalWritten, bytesWritten)
-		atomic.AddUint64(&blockCount, 1)
+		totalWritten.Add(bytesWritten)
```

Comment:
I guess everyone prefers go 1.19 syntax but me :P It makes sense.

Reply:
Yeah, sorry, not really a preference, I only touched these because I found new race conditions with the use of these values at the very end—they needed to be loaded; I don't recall why it _felt_ easier to switch it out to new style 🤷